cognite/neat/_utils/migration.py

```diff
@@ -0,0 +1,154 @@
+import typing
+from pprint import pprint
+from typing import Any, TypeVar, cast
+
+import pandas as pd
+from cognite.client import CogniteClient
+from cognite.client.data_classes import Asset, Event, FileMetadata, Sequence, TimeSeries
+from cognite.client.data_classes.data_modeling import Node, ViewId
+
+from cognite.neat._rules._constants import get_reserved_words
+
+from .auxiliary import local_import
+
+T_Classic = TypeVar("T_Classic", bound=Asset | TimeSeries | Event | Sequence | FileMetadata)
+
+
+SERVER_SIDE = {"parentId", "id", "createdTime", "lastUpdatedTime", "rootId", "source", "dataSetId"}
+NODE_SERVER_SIDE = {"version", "lastUpdatedTime", "createdTime", "instanceType", "path", "root", "pathLastUpdatedTime"}
+RENAMING = {"parent": "parentExternalId"}
+RESERVED_PROPERTIES = {word.replace("_", "").lower() for word in get_reserved_words("property")}
+
+
+def display_diffs(
+    source_items: typing.Sequence[Asset | TimeSeries | Event | Sequence | FileMetadata],
+    target_view: ViewId | tuple[str, str, str],
+    source: CogniteClient,
+    target: CogniteClient | None = None,
+) -> None:
+    """Displays the differences between the source and target objects."""
+    local_import("deepdiff", "compare")
```

Comment:
Its a poor pattern to call into the import machinery deep within functions in Python - just because we can do it, doesn't mean we should.

There is also a developer affordance impact of having custom import layers - IDEs and other code level tools don't typically handle things that do imports but aren't import statements well.

I do see that you need this in some places, to switch to fallback codepaths, but that isn't the case here for anything in this module. -> If this function called from the contexts where you don't have `deepdiff` available, it will fail at runtime.

So, please either:
- document the failure modes in the docstring, including what it will raise
- or lift the import out of the function so that code which has imported the function knows it is usable.

Reply:
This is an internal function and we do not want to add an extra dependency on DeepDiff to everyone that uses neat, thus we use a local import and mark this as an optional dependency. The local_import function raises an appropriate error if the dependency is missing.